File: Python3/path_sum_III.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def pathSum(self, root: TreeNode, sum: int) -> int:
        counts = [0]
        
        def helper(root, sums):
            if not root: return
            sums = [s + root.val for s in sums] + [root.val]
            counts[0] += sums.count(sum)
            helper(root.left, sums)
            helper(root.right, sums)
        
        helper(root, [])
        
        return counts[0]

File: C++/path_sum_III.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int counts = 0;
    
    void helper(TreeNode* root, int sum) {
        if(!root) return;
        if(root->val == sum) counts++;
        helper(root->left, sum-root->val);
        helper(root->right, sum-root->val);
    }
    
    int pathSum(TreeNode* root, int sum) {
        if(root) {
            helper(root, sum);
            pathSum(root->left, sum);
            pathSum(root->right, sum);
        }
        return counts;
    }
};
